src/engine/docker_engine.rs

```diff
@@ -43,6 +43,35 @@ impl DockerEngine {
 
         let config = ApplicationConfig::from_env();
 
+        // Create the network if needed
+        let mut filters: HashMap<&str, Vec<&str>> = HashMap::new();
+        filters.insert("name", vec![&config.traefik_config.network_name]);
+        let list_options: ListNetworksOptions<&str> = ListNetworksOptions { filters };
+
+        let result_networks = docker.list_networks(Some(list_options)).await;
+
+        if result_networks.is_err() {
+            error!("Couldn't list networks !")
+        } else if result_networks.is_ok() && result_networks.unwrap_or_default().len() == 0 {
+            // Create the fallback network
+            let network_options: CreateNetworkOptions<&str> = CreateNetworkOptions {
+                name: &config.traefik_config.network_name,
+                ..Default::default()
+            };
+
+            let result_network_creation = docker.create_network(network_options).await;
+            match result_network_creation {
+                Ok(_) => trace!(
+                    "Docker network created: {}",
+                    &config.traefik_config.network_name
+                ),
+                Err(_) => error!(
+                    "Failed to create Docker network: {}",
+                    &config.traefik_config.network_name
+                ),
+            }
+        }
```

Comment:
Something like that would be more idiomatic: (idk if that compiles too; also, note how the error is handled differently)
```suggestion
        match docker.list_networks(Some(list_options))
            .await
        {
            Ok(v) if v.len() == 0 => {
                // Create the fallback network
                let network_options: CreateNetworkOptions<&str> = CreateNetworkOptions {
                    name: &config.traefik_config.network_name,
                    ..Default::default()
                };
    
                let result_network_creation = docker.create_network(network_options).await;
                match result_network_creation {
                    Ok(_) => trace!(
                        "Docker network created: {}",
                        &config.traefik_config.network_name
                    ),
                    Err(_) => error!(
                        "Failed to create Docker network: {}",
                        &config.traefik_config.network_name
                    ),
                }
            },
            Ok(_) => (),
            Err(e) => {
                Err(e).context("Failed to list Docker networks")
            },
        }
```

Reply:
I did something similar at first, but I strongly disliked the additional indentation that comes with it.
I don't have the force to object back much at your suggestion, but I would appreciate something that doesn't me feel like nesting if else statements